src/NextjsBase.ts

```diff
@@ -72,6 +72,13 @@ export interface NextjsBaseProps {
    * If omitted, the layer will be created.
    */
   readonly sharpLayerArn?: string;
+
+  /**
+   * By default all CloudFront cache will be invalidated on deployment.
+   * This can be set to true to skip the full cache invalidation, which
+   * could be important for some users.
+   */
+  readonly skipFullInvalidation?: boolean;
```

Comment:
So has this never worked? Any files in public/ that were changed but had same name wouldn't get updated in CloudFront cache?

Also, are we sure we want default full invalidation? It makes sense to me, but I want to make sure. @revmischa, can you provide any insight into history of this?

Reply:
It may have not been hooked up by default, as I recall I had some circular dependency issues here.
It's fine to do a full invalidation, I don't think there would be any problem with that. I was just thinking that since the NextJS static file names have hashes in them, it probably isn't strictly necessary to invalidate the cache for static files since the files that change would get new names. Probably better to just invalidate `/*`

Reply:
I'm _assuming_ that this worked before because the rewriter invalidates the paths in the public S3 bucket. I'm not 100% on this, but I would imagine it would have been called out if it wasn't working before (or perhaps people just were happy creating invalidations in CloudFront manually?).

What I am certain of though is that in my testing at least, without this invalidation the CloudFront cache for SSG pages was not being cleared on deployment.

As for whether the default is appropriate, or whether there needs to be a flag at all... I could be easily convinced either way. On one hand, my honest opinion is somebody using this for a larger, complex project will likely want a more carefully planned and executing cache flushing policy / system, so it's likely some modification of this would happen anyway, so perhaps adding the "skipFullInvalidation" flag is extra noise. On the other hand, it's possible some people would like the option of quickly disabling the flush so they can do it manually and having this config option might be handy.

Reply:
Agreed. Let's keep it.